### PR TITLE
Fix slicing dimensions in rotary embeddings

### PR DESCRIPTION
**Pull request by ploshkin**

### CODE REVIEW

File: flash_attn/layers/rotary.py [hunk 1]
```diff
@@ -43,13 +43,11 @@ def forward(ctx, x, cos, sin, inplace=False):
         rotary_dim *= 2
         assert rotary_dim <= headdim
         assert seqlen <= rotary_seqlen
-        assert cos.shape == (rotary_seqlen, rotary_dim // 2)
-        assert sin.shape == (rotary_seqlen, rotary_dim // 2)
```
Comment (tridao) [inline flash_attn/layers/rotary.py:47]: Can we keep the shape checking for sin?

File: flash_attn/layers/rotary.py [hunk 2]
```diff
@@ -92,14 +90,12 @@ def forward(ctx, qkv, cos, sin):
         rotary_dim *= 2
         assert rotary_dim <= headdim
         assert seqlen <= rotary_seqlen
-        assert cos.shape == (seqlen, rotary_dim // 2)
-        assert sin.shape == (seqlen, rotary_dim // 2)
```
Comment (tridao) [inline flash_attn/layers/rotary.py:96]: Can we keep the shape checking for sin?

**Review by tridao — CHANGES_REQUESTED** [review]
Thanks for the fix @ploshkin!

File: flash_attn/layers/rotary.py (hunk 1 above)
Comment (ploshkin) [reply flash_attn/layers/rotary.py:47]: Ah, sure

File: flash_attn/layers/rotary.py (hunk 2 above)
Comment (ploshkin) [reply flash_attn/layers/rotary.py:96]: Here I changed the assertion because length of sin might be greater than `seqlen` in general

Comment (tridao) [reply flash_attn/layers/rotary.py:96]: Oh right, that's a great point.

**Review by tridao — APPROVED** [review]
Perfect, thanks again @ploshkin!

**Pull request merged**